constantine/ethereum_eip7594_kzg.nim

```diff
@@ -0,0 +1,196 @@
+    case status
+    of cttCodecScalar_Success:                          discard
+    of cttCodecScalar_Zero:                             discard
+    of cttCodecScalar_ScalarLargerThanCurveOrder:       result = cttEthKzg_ScalarLargerThanCurveOrder; break Section
+
+template check(Section: untyped, evalExpr: CttCodecEccStatus): untyped {.dirty.} =
+  # Translate codec status code to KZG status code
+  # Exit current code block
+  block:
+    let status = evalExpr # Ensure single evaluation
+    case status
+    of cttCodecEcc_Success:                             discard
+    of cttCodecEcc_InvalidEncoding:                     result = cttEthKzg_EccInvalidEncoding; break Section
+    of cttCodecEcc_CoordinateGreaterThanOrEqualModulus: result = cttEthKzg_EccCoordinateGreaterThanOrEqualModulus; break Section
+    of cttCodecEcc_PointNotOnCurve:                     result = cttEthKzg_EccPointNotOnCurve; break Section
+    of cttCodecEcc_PointNotInSubgroup:                  result = cttEthKzg_EccPointNotInSubGroup; break Section
+    of cttCodecEcc_PointAtInfinity:                     discard
+
+
+# from eip4844, import
+# bytes_to_bls_field
+# bls_field_to_bytes
+
+func cell_to_coset_evals(evals: var CosetEvals,cell: Cell): CttCodecEccStatus=
+  # Convert an untrusted ``Cell`` into a trusted ``CosetEvals``
+  var 
+    start: int
+    ending: int
+    value:Fr[BLS12_381]
+
+  let view = cast[ptr array[FIELD_ELEMENTS_PER_CELL, array[32, byte]]](cell.unsafeAddr)
+  for i in 0..FIELD_ELEMENTS_PER_CELL:
+    start = i*BYTES_PER_FIELD_ELEMENT
+    ending = (i+1)*BYTES_PER_FIELD_ELEMENT
+    var status= value.bytes_to_bls_field(view[i])
+    if status notin {cttCodecScalar_Success, cttCodecScalar_Zero}:
+      return cttCodecEcc_PointNotOnCurve
+    evals[i]=value
+
+  return cttCodecEcc_Success
+
+
+
+func coset_evals_to_cell(dst: var Cell,coset_evals: CosetEvals): CttCodecEccStatus=
+    # Convert a trusted ``CosetEval`` into an untrusted ``Cell``.
+    # for i in 0 ..< FIELD_ELEMENTS_PER_CELL:
+    #     var bytes = bls_field_to_bytes(cosetEvals[i])
+    #     for j in 0 ..< bytes.len:
+    #         dst[i* bytes.len + j] = bytes[j]
+
+    return  cttCodecEcc_Success
+
+
+
+func compute_kzg_proof_multi_impl(dst:var (KZGProof, CosetEvals),poly_coeff: PolyCoeff,zs: Coset)=
+    # Compute a KZG multi-evaluation proof for a set of `k` points.
+
+    # This is done by committing to the following quotient polynomial:
+    #     Q(X) = f(X) - I(X) / Z(X)
+    # Where:
+    #     - I(X) is the degree `k-1` polynomial that agrees with f(x) at all `k` points
+    #     - Z(X) is the degree `k` polynomial that evaluates to zero on all `k` points
+
+    # We further note that since the degree of I(X) is less than the degree of Z(X),
+    # the computation can be simplified in monomial form to Q(X) = f(X) / Z(X)
+    
+
+    # For all points, compute the evaluation of those points
+
+    for i in z:
+      ys[i]=evaluate_polynomialcoeff(polynomial_coeff, z[i])
```

Comment:
As this is not defined, I assume this is pseudocode from the spec so I'll stop reviewing the file here.

It's OK to delete those, only check in the coset_eval_to_cell and the 8k lines of test vectors.